File: php/src/Model/WorkerAvailabilityState.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A worker's own shift switch, after flipping it.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerAvailabilityState
{
    public function __construct(
        public readonly string $workerId,
        public readonly bool $available,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            available: (bool) ($data['available'] ?? false),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerAvailabilityState.java
package io.fivexer.sdk.model;

/**
 * A worker's own shift switch, after flipping it.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerAvailabilityState {
    private String workerId;
    private Boolean available;

    public String getWorkerId() { return workerId; }
    public Boolean getAvailable() { return available; }
}
